### ISSUE TITLE
Failing way too quickly can break the fail screen and replay feature

### ISSUE THREAD

**Issue by GPJmp** [ppy/osu-stable-issues#496]
I've noticed that failing too quickly can cause the failing animation to be skipped in the replay, the replay will finish with a broken ranking screen and it can even be saved

How to replicate it:
- Make a beatmap with high HP (Preferably 10)
- Make a stack of about 6 circles
- Play the map, but click on each circle too early so that you miss them on purpose
Then the fail screen will show up midway through the circle falling animation, and the replay breaks. On the ranking screen the graph will be cut off where you failed, and the numbers will be incorrect

[Video showing the glitch](https://youtu.be/BTMmGSXZ890)

**Comment by peppy** [comment]
Is this actually reproducible in normal gameplay? I don't think it's worth fixing if you have to have such a short map to reproduce in the first place.

**Comment by GPJmp** [comment]
Yeah, it is. I found this bug because of a ranked beatmap where I could reproduce it multiple times on the same part, and few other players also experienced it on different maps. Just thought I'd report it, even though it's not really worth fixing
